core/src/io/anuke/mindustry/ui/dialogs/CustomRulesDialog.java

```diff
@@ -27,15 +29,22 @@ public CustomRulesDialog(){
     public void show(Rules rules, Gamemode gamemode){
         this.rules = rules;
         this.selectedGamemode = gamemode;
-        show();
+        super.show();
+    }
+
+    @Override
+    public Dialog show(){
```

Comment:
Don't override `show()`, register a listener for it instead, e.g. `shown(() -> this.rules = state.rules);`.